alchemist-project.el

```diff
@@ -125,16 +125,45 @@ completion will be work.
     (when project-root
       (setq default-directory project-root))))
 
+(defun alchemist-toggle-file-and-tests-other-window ()
+  "Toggle between a file and its tests in other window."
+  (interactive)
+  (if (alchemist--is-test-file-p)
+      (alchemist--project-open-file-for-current-tests 'find-file-other-window)
+    (alchemist--project-open-tests-for-current-file 'find-file-other-window)))
+
+(defun alchemist-toggle-file-and-tests ()
+  "Toggle between a file and its tests in the current window."
+  (interactive)
+  (if (alchemist--is-test-file-p)
+      (alchemist--project-open-file-for-current-tests 'find-file)
+    (alchemist--project-open-tests-for-current-file 'find-file)))
+
+(defun alchemist--is-test-file-p ()
+  "Check wether the visited file is a test file."
+  (string-match "_test\.exs$" (buffer-file-name)))
+
+(defun alchemist--project-open-file-for-current-tests (toggler)
+  "Open the appropriate implementation file for the current buffer by calling TOGGLER with filename."
+  (let* ((filename (file-relative-name (buffer-file-name) (alchemist-project-root)))
+         (filename (replace-regexp-in-string "^test/" "lib/" filename))
+         (filename (replace-regexp-in-string "_test\.exs$" "\.ex" filename))
+         (filename (format "%s/%s" (alchemist-project-root) filename)))
+    (funcall toggler filename)))
+
 (defun alchemist-project-open-tests-for-current-file ()
-  "Opens the appropriate test file for the current buffer file
-in a new window."
+  "Open the appropriate test file for the current buffer file in a new window."
+  (alchemist--project-open-tests-for-current-file 'find-file-other-window))
+
+(defun alchemist--project-open-tests-for-current-file (toggler)
```

Comment:
I've left the function `private` because I wanted to provide both `alchemist-toggle-file-and-tests{,-other-window}`. In the normal version, the `alchemist--project-open-tests-for-current-file` is called with `'find-file` and in the `*-other-window` version, it is called with `'find-file-other-window`.